animations-dbus/animations-dbus-server-animation-manager.h

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2018 Endless Mobile, Inc.
+ *
+ * animations-dbus is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * libanimation-dbus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with libanimation-dbus.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <animations-dbus-server-effect.h>
+#include <animations-dbus-server-effect-factory-interface.h>
+
+#include <gio/gio.h>
+#include <glib.h>
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+#define ANIMATIONS_DBUS_TYPE_SERVER_ANIMATION_MANAGER animations_dbus_server_animation_manager_get_type ()
+G_DECLARE_FINAL_TYPE (AnimationsDbusServerAnimationManager, animations_dbus_server_animation_manager, ANIMATIONS_DBUS, SERVER_ANIMATION_MANAGER, AnimationsDbusAnimationManagerSkeleton)
+
+typedef struct _AnimationsDbusServer AnimationsDbusServer;
```

Comment:
This and some other places generate a warning because of redefinition.
I think you're trying to simulate a forward declaration, but I guess the way to go is to have a 3rd file `animation-dbus-server-type.h` that you can include in both? Analogously for other similar warnings?

Reply:
Yeah, I think that would solve the problem. Thanks for the tip!